editor/editor_node.cpp

```diff
@@ -2666,6 +2666,13 @@ void EditorNode::_menu_option_confirm(int p_option, bool p_confirmed) {
 		case FILE_CLOSE: {
 			_scene_tab_closed(editor_data.get_edited_scene());
 		} break;
+		case FILE_SAVE_SCENE_SILENTLY: {
+			Node *scene = editor_data.get_edited_scene_root();
+			if (scene && !scene->get_scene_file_path().is_empty() && DirAccess::exists(scene->get_scene_file_path().get_base_dir())) {
+				_save_scene(scene->get_scene_file_path());
+				save_editor_layout_delayed();
+			}
+		} break;
```

Comment:
Add a comment to explain why this is needed:

```suggestion
		case FILE_SAVE_SCENE_SILENTLY: {
			// Save scene without displaying progress dialog. Used to work around
			// errors when Save on Focus Loss kicks in.
			Node *scene = editor_data.get_edited_scene_root();
			if (scene && !scene->get_scene_file_path().is_empty() && DirAccess::exists(scene->get_scene_file_path().get_base_dir())) {
				_save_scene(scene->get_scene_file_path());
				save_editor_layout_delayed();
			}
		} break;
```